pipeline/runtime/runtestsV2.go

```diff
@@ -0,0 +1,330 @@
+			RunOnlySelectedTests = false // TI selected all the tests to be run
+			return selection, false
+		}
+
+		if lastSuccessfulCommitID == "" {
+			log.Infoln("Test Intelligence determined to run all the tests to bootstrap")
+			RunOnlySelectedTests = false // TI selected all the tests to be run
+			return selection, false
+		}
+
+		log.Infoln("Using reference commit: ", lastSuccessfulCommitID)
+		files, err = utils.GetChangedFilesPush(ctx, workspace, lastSuccessfulCommitID, tiConfig.GetSha(), log)
+		if err != nil {
+			log.Errorln("Unable to get changed files list. Running all the tests.", "error", err)
+			RunOnlySelectedTests = false
+			return selection, false
+		}
+	} else {
+		files, err = utils.GetChangedFilesPR(ctx, workspace, log)
+		if err != nil || len(files) == 0 {
+			log.Errorln("Unable to get changed files list for PR. Running all the tests.", "error", err)
+			RunOnlySelectedTests = false
+			return selection, false
+		}
+	}
+
+	filesWithpkg := java.ReadPkgs(log, fs, workspace, files)
+	selection, err = utils.SelectTests(ctx, workspace, filesWithpkg, RunOnlySelectedTests, stepID, fs, tiConfig)
+	if err != nil {
+		log.WithError(err).Errorln("There was some issue in trying to figure out tests to run. Running all the tests")
+		RunOnlySelectedTests = false
+	} else if selection.SelectAll {
+		log.Infoln("Test Intelligence determined to run all the tests")
+		RunOnlySelectedTests = false // TI selected all the tests to be run
+	} else {
+		log.Infoln(fmt.Sprintf("Running tests selected by Test Intelligence: %s", selection.Tests))
+	}
+
+	return selection, true
+}
+
+func createJavaConfigFile(tmpDir string, fs filesystem.FileSystem, log *logrus.Logger) (string, error) {
+
+	dir := fmt.Sprintf(outDir, tmpDir)
+	err := fs.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create nested directory %s", dir))
+		return "", err
+	}
+
+	iniFileDir := fmt.Sprintf("%s/new", tmpDir)
+	err = fs.MkdirAll(iniFileDir, os.ModePerm)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create nested directory %s", iniFileDir))
+		return "", err
+	}
+	iniFile := fmt.Sprintf("%s/config.ini", iniFileDir)
+	data := fmt.Sprintf(`outDir: %s
+	logLevel: 0 
+	logConsole: false
+	writeTo: JSON
+	packageInference: true`, dir)
+
+	log.Infof("Attempting to write to %s with config:\n%s", iniFile, data)
+	f, err := fs.Create(iniFile)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create config file %s", iniFile))
+		return "", err
+	}
+
+	_, err = f.WriteString(data)
+	defer f.Close()
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not write %s to config file %s", data, iniFile))
+		return "", err
+	}
+
+	return iniFile, nil //path of config.ini file
+}
+
+func getPreCmd(tmpFilePath string, fs filesystem.FileSystem, log *logrus.Logger) (string, error) {
+	var preCmd string
+	iniFilePath, err := createJavaConfigFile(tmpFilePath, fs, log)
```

Comment:
Nit: `configFilePath`

Reply:
tmpFilePath is being provided and tiConfig. This is /tmp/engine